src/script/descriptor.cpp

```diff
@@ -1143,36 +1143,46 @@ std::unique_ptr<DescriptorImpl> ParseScript(uint32_t& key_exp_index, Span<const
         ++key_exp_index;
         std::vector<std::unique_ptr<DescriptorImpl>> subscripts;
         std::vector<int> depths;
-        while (expr.size()) {
+        if (expr.size()) {
+            std::vector<bool> counts; //!< whether we already had an expression at that depth
             if (!Const(",", expr)) {
                 error = strprintf("tr: expected ',', got '%c'", expr[0]);
                 return nullptr;
             }
-            auto sarg = Expr(expr);
-            auto colon = std::find(sarg.begin(), sarg.end(), ':');
-            if (colon == sarg.end()) {
-                error = "tr() requires scripts with 'depth:' prefix";
-                return nullptr;
-            }
-            uint32_t depth;
-            if (!ParseUInt32(std::string(sarg.begin(), colon), &depth)) {
-                error = "depth is not numeric in tr()";
-                return nullptr;
-            }
-            if (depth > TAPROOT_CONTROL_MAX_NODE_COUNT) {
-                error = strprintf("depth is out of range in tr() (0 <= depth <= %i)", TAPROOT_CONTROL_MAX_NODE_COUNT);
-                return nullptr;
+            while (expr.size()) {
+                if (Const("{", expr)) {
+                    counts.push_back(false);
+                    if (counts.size() > TAPROOT_CONTROL_MAX_NODE_COUNT) {
+                        error = strprintf("tr() supports at most %i nesting levels", TAPROOT_CONTROL_MAX_NODE_COUNT);
+                        return nullptr;
+                    }
+                    continue;
+                }
+                auto sarg = Expr(expr);
+                auto subdesc = ParseScript(key_exp_index, sarg, ParseScriptContext::P2TR, out, error);
```

Comment:
you could directly emplace_back the subdesc and then nullcheck/return nullptr, rather than needing to std::move it.

Reply:
Done.